src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/OrderingHelper.cs

```diff
@@ -0,0 +1,320 @@
+            {
+                // Grab the project.
+                var project = await writeLock.GetProjectAsync(configuredProject).ConfigureAwait(true);
+
+                // We must perform a checkout of the project file before we can modify it.
+                await writeLock.CheckoutAsync(project.FullPath).ConfigureAwait(true);
+
+                return Move(configuredProject.UnconfiguredProject, project, projectTree, moveAction);
+            }
+        }
+
+        /// <summary>
+        /// Checks if the given project tree can move up over one of its siblings.
+        /// </summary>
+        public static bool CanMoveUp(IProjectTree projectTree)
+        {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return GetSiblingByMoveAction(projectTree, MoveAction.Up) != null;
+        }
+
+        /// <summary>
+        /// Move the project tree up over one of its siblings.
+        /// </summary>
+        public static bool MoveUp(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return Move(unconfiguredProject, project, projectTree, MoveAction.Up);
+        }
+
+        /// <summary>
+        /// Move the project tree up over one of its siblings.
+        /// </summary>
+        public static Task<bool> MoveUpAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
+        {
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return MoveAsync(configuredProject, projectTree, MoveAction.Up);
+        }
+
+        /// <summary>
+        /// Checks if the given project tree can move down over one of its siblings.
+        /// </summary>
+        public static bool CanMoveDown(IProjectTree projectTree)
+        {
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return GetSiblingByMoveAction(projectTree, MoveAction.Down) != null;
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static bool MoveDown(UnconfiguredProject unconfiguredProject, Project project, IProjectTree projectTree)
+        {
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
+            Requires.NotNull(project, nameof(project));
+            Requires.NotNull(projectTree, nameof(projectTree));
+
+            return Move(unconfiguredProject, project, projectTree, MoveAction.Down);
+        }
+
+        /// <summary>
+        /// Move the project tree down over one of its siblings.
+        /// </summary>
+        public static Task<bool> MoveDownAsync(ConfiguredProject configuredProject, IProjectTree projectTree)
```

Comment:
Can we re-order these to be public -> private so the entry point into this class is at the top?

Reply:
Yup.

Reply:
done.